pilot/pkg/kube/inject/inject.go

```diff
@@ -187,15 +187,27 @@ func validateCIDRList(cidrs string) error {
 	return nil
 }
 
-func validatePortList(ports string) error {
-	if len(ports) > 0 {
-		for _, port := range strings.Split(ports, ",") {
-			if _, err := strconv.ParseInt(port, 10, 16); err != nil {
-				return fmt.Errorf("failed parsing port '%s': %v", port, err)
+func splitPorts(portsString string) []string {
+	return strings.Split(portsString, ",")
+}
+
+func parsePorts(portsString string) ([]int, error) {
+	if len(portsString) > 0 {
+		ports := make([]int, len(portsString))
+		for i, portStr := range splitPorts(portsString) {
+			port, err := strconv.ParseInt(portStr, 10, 16)
```

Comment:
you probably want parseUint here. parseInt will allow negatives and reject valid port values > 32767.
also, do you need to trim space in case the string list has spaces?

Reply:
done.